graphql_compiler/schema_generation/schema_graph.py

```diff
@@ -509,84 +470,59 @@ def _set_up_schema_elements_of_kind(self, class_name_to_definition, kind, class_
                                          u'more than once, this is not allowed!'
                                          .format(property_name, class_name))
 
-                property_descriptor = self._create_descriptor_from_property_definition(
-                    class_name, property_definition, class_name_to_definition)
-
-                if property_name in allowed_duplicated_edge_property_names:
-                    links[property_name].append(property_descriptor)
+                if property_name in set(links.keys()):
+                    self._validate_link(orientdb_property_definition, class_name,
+                                        class_name_to_definition)
+                    links[property_name].append(orientdb_property_definition['linkedClass'])
                 else:
+                    property_descriptor = self._create_descriptor_from_orientdb_property_definition(
+                        class_name, orientdb_property_definition)
                     property_name_to_descriptor[property_name] = property_descriptor
 
-            for property_name in allowed_duplicated_edge_property_names:
-                elements = {
-                    property_descriptor.qualifier
-                    for property_descriptor in links[property_name]
-                }
+            self._elements[class_name] = SchemaElement(class_name, kind, abstract,
+                                                       property_name_to_descriptor, class_fields)
+            for link_direction in links.keys():
```

Comment:
nit: just iterate the keys and values of the dict, you're using the value anyway. This way I won't have to do mental gymnastics to figure out if `link_direction` is in `links` when you access it.